File: aeongseo/BOJ/CPP/boj10845.cpp
/*** 10845. 큐 ***/

#include<iostream>
#include<string>
#include<deque>
using namespace std;

int main() {
	int N, num;
	cin >> N;
	string order;
	deque<int> arr;	// 덱 사용
	for (int i = 0; i < N; i++) {
		cin >> order;
	
		if (order == "push") {	// push일 때 숫자 입력받고 enque
			cin >> num;
			arr.push_back(num);
			//cout << arr.back() << '\n';
		}
		else if (order == "pop") { // pop일 때 덱이 비었으면 -1 출력, 비어있지 않다면 맨앞 숫자 출력 후 deque
			if (arr.empty()) {
				cout << -1 << '\n';
			}
			else {
			cout << arr.front() << '\n';
			arr.pop_front();
			}
		}
		else if (order == "size") { //size일 때 덱의 요소 개수 출력
			cout << arr.size() << '\n';
		}
		else if (order == "empty") {	// empty일 때 덱이 비었으면 1 출력, 비어있지 않다면 0 출력
			if (arr.empty()) {
				cout << 1 << '\n';
			}
			else {
				cout << 0 << '\n';
			}
		}
		else if (order == "front") {	// front일 때 덱이 비었다면 -1 출력, 비어있지 않다면 맨앞 숫자 출력
			if (arr.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << arr.front() << '\n';
			}
		}
		else if (order == "back") {	// back일 때 덱이 비었다면 -1 출력, 비어있지 않다면 맨뒤 숫자 출력
			if (arr.empty()) {
				cout << -1 << '\n';
			}
			else {
				cout << arr.back() << '\n';
			}
		}

	}
	return 0;
}

File: LSH0707/BOJ/boj10845.py
import sys
input = sys.stdin.readline
from collections import deque
N = int(input())
arr = [list(map(str, input().split())) for _ in range(N)]
ans = deque()
for i in range(N):  # 리스트 순회하며 조건에 맞는 값 출력
    if arr[i][0] == 'push':
        ans.append(int(arr[i][1]))
    elif arr[i][0] == 'pop':
        if len(ans) > 0:
            print(ans.popleft())
        else:
            print(-1)
    elif arr[i][0] == 'size':
        print(len(ans))
    elif arr[i][0] == 'empty':
        if len(ans) == 0:
            print(1)
        else:
            print(0)
    elif arr[i][0] == 'front':
        if len(ans) > 0:
            print(ans[0])
        else:
            print(-1)
    elif arr[i][0] == 'back':
        if len(ans) > 0:
            print(ans[-1])
        else:
            print(-1)